fdbserver/ApplyMetadataMutation.h

```diff
@@ -22,14 +22,54 @@
 #define FDBSERVER_APPLYMETADATAMUTATION_H
 #pragma once
 
+#include <cstddef>
+
 #include "fdbclient/BackupAgent.actor.h"
 #include "fdbclient/MutationList.h"
 #include "fdbclient/Notified.h"
+#include "fdbclient/StorageServerInterface.h"
 #include "fdbclient/SystemData.h"
 #include "fdbserver/IKeyValueStore.h"
 #include "fdbserver/LogProtocolMessage.h"
 #include "fdbserver/LogSystem.h"
 #include "fdbserver/ProxyCommitData.actor.h"
+#include "flow/FastRef.h"
+
+// Resolver's data for applyMetadataMutations() calls.
+struct ResolverData {
+	const UID dbgid;
+	IKeyValueStore* txnStateStore = nullptr;
+	KeyRangeMap<ServerCacheInfo>* keyInfo = nullptr;
+	Arena arena;
+	// Whether configuration changes. If so, a recovery is forced.
+	bool& confChanges;
```

Comment:
it's weird that this is a reference, not a pointer

Reply:
Yes, I changed to be a pointer.